File: de.unistuttgart.iste.sqa.mpw.hamstersimulator.java/core/src/main/java/de/unistuttgart/hamster/mpw/impl/GameLogImpl.java
package de.unistuttgart.hamster.mpw.impl;

import de.unistuttgart.hamster.mpw.Actor;
import de.unistuttgart.hamster.mpw.GameLog;
import de.unistuttgart.hamster.mpw.LogEntry;

public class GameLogImpl extends GameLog {

	@Override
	public void write(Actor actor, String message) {
		var logEntry = new LogEntry();
		logEntry.setActor(actor);
		logEntry.setMessage(message);
		addToLogEntries(logEntry);
	}

}

File: de.unistuttgart.iste.sqa.mpw.framework.cpp/core/src/impl/GameLogImpl.cpp
#include "GameLogImpl.h"

#include "LogEntry.h"
#include <stdexcept>

namespace mpw {

GameLogImpl::GameLogImpl() {
}

void GameLogImpl::write(std::shared_ptr<mpw::Actor> actor, std::string message) {
    auto entry = std::make_shared<LogEntry>();
    entry->setActor(actor);
    entry->setMessage(message);
	addToLogEntries(entry);
}

}

